### DISCUSSION TITLE
patternProperties does not take the $ref as type

### DISCUSSION THREAD

**Discussion by ag0n1k** [koxudaxi/datamodel-code-generator#469] [Q&A] [answered]
Hi,

Try to use docker-compose jsonschema located at https://github.com/docker/cli/blob/master/cli/compose/schema/data/config_schema_v3.7.json and there is some filed with patternProperties which indicates the class of "child":
```
"services": {
      "id": "#/properties/services",
      "type": "object",
      "patternProperties": {
        "^[a-zA-Z0-9._-]+$": {
          "$ref": "#/definitions/service"
        }
      },
      "additionalProperties": false
    },
```

The generator missed this and place "Any" there:
```
class Model(BaseModel):
    version: str
    services: Optional[Dict[str, Any]] = None
    ...
```

Is this a bug or I have not checked some keys on cli to fill it right?

**Comment by koxudaxi** [comment, marked as answer]
@ag0n1k 
I'm sorry for the late comment.
The older version doesn't support `patternProperties`
But, I have released a new version `0.11.12` which supports `patternProperties`
https://github.com/koxudaxi/datamodel-code-generator/issues/470

**Reply by ag0n1k** [reply]
For the record: I think the update not just minor improvement. For example the diff beetween 0.11.12 and 0.11.8 is huge for the compose model (https://github.com/docker/cli/blob/master/cli/compose/schema/data/config_schema_v3.7.json) and don't think there is some backwards compatibility.

**Reply by koxudaxi** [reply]
@ag0n1k 
Thank you for the comment.
Should be the feature provided as a CLI option?
What do you think about it?
I would like to cover many use-cases.

**Reply by ag0n1k** [reply]
I guess the main problem not with the 'new' features, which also breaks bacward compatibility, but in case that for now there always an option which guaranties that there will be `Extra.allow` or `Extra.forbid` and there is one way to provide other Config parameters - use custom templates which conflicts with auto-assignment for only `extra`. 

I think that the `extra` must be the one of the many which can be also provided like:
```
{
  "#all#": {
    "additionalProperties": true,
    "use_enum_values": true,
    "validate_all": true,
    "validate_assignment": true,
    "allow_mutation": false,
    "frozen": true
  }
}
```
it would be great!

**Reply by ag0n1k** [reply]
btw the `{"#all#": {"additionalProperties": true}}` found only with debug =(
